### ISSUE TITLE
Remove bower in favor of npm

### ISSUE THREAD

**Issue by sloria** [sloria/cookiecutter-flask#56]
Bower is [no longer maintained](https://github.com/bower/bower/pull/1748#issuecomment-150104311), and is [redundant with npm](https://www.quora.com/Why-use-Bower-when-there-is-npm).

**Comment by barraponto** [comment]
You mean use browserify instead of bower?
As far as I can see, bower is still the most popular tool...

**Comment by sloria** [comment]
No, bower and npm are package managers, but npm is the preferred solution at this time (see https://www.quora.com/Why-use-Bower-when-there-is-npm). 

Browserify is a build tool.

**Comment by mjsisley** [comment]
+1 on npm

**Comment by barraponto** [comment]
@sloria i know, but afaik npm handles node-specific backend

**Comment by sloria** [comment]
@barraponto Correct; part of this change would be integrating a build tool like webpack to make the packages work on the front-end.

**Comment by mjsisley** [comment]
FWIW I am using npm and webpack in order to use React in a cookiecutter-django project and getting it set up was fairly straightforward.

**Comment by tracker1** [comment]
@barraponto `npm` is a JavaScript package manager, and while there are some conventions for using node-style modules, and proposals for es6-style modules, you can still use direct scripts from `node_modules/foo-package/bin/foo-release-x.x.x.min.js` if you choose to, this is how non-js resources are accessed, and copied in, similar to with bower_modules.

**Comment by barraponto** [comment]
I see. I don't know about you, @tracker1, but I usually avoid exposing `node_modules` over http. Tools like webpack and browserify read the files from that folder and build output files which are ok for public access. That includes css and other non-js resources.

In any case, I guess my position here is: let's make the switch and add a builder setup. New users will welcome the simplicity of running a command and having everything work, while advanced users will know how to remove webpack with minimal effort.

**Comment by sbhtw** [comment]
@barraponto The template uses [Flask-Assets](https://flask-assets.readthedocs.io/en/latest/) and all source files (js/css) are automatically preprocessed, bundled and copied, just like webpack or other build tools would work.

See `flask assets` on CLI and generated `assets.py`

I know that Webpack & Co are superior than any non-JavaScript tool even can be but this would complicate the setup.

So I would opt for migration to NPM but no JavaScript build tool.

I know that webpack & co are superior than any non-JS tool even can be but this would complicate the setup.

**Comment by barraponto** [comment]
@sbhtw i'm ok with that, as long as flask-assets ships configured to use stuff out of npm_modules.

**Comment by jacobSingh** [comment]
Hey! I'm trying to figure out the current setup here and am confused.

I don't see how bower is even being used.  When I run bower install, it puts the packages in bower_components, but the code in static/libs/js isn't a link to that code...

I tried to add highhcarts, and now I get this:

bower                            error Unexpected string in JSON at position 497

Stack trace:
SyntaxError: Unexpected string in JSON at position 497
    at Object.parse (native)
    at /usr/local/lib/node_modules/bower/lib/core/Manager.js:209:37
    at _fulfilled (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:796:13)
    at /usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:604:44
    at runSingle (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

Console trace:
Error
    at StandardRenderer.error (/usr/local/lib/node_modules/bower/lib/renderers/StandardRenderer.js:81:37)
    at Logger.<anonymous> (/usr/local/lib/node_modules/bower/lib/bin/bower.js:110:26)
    at emitOne (events.js:96:13)
    at Logger.emit (events.js:188:7)
    at Logger.emit (/usr/local/lib/node_modules/bower/lib/node_modules/bower-logger/lib/Logger.js:29:39)
    at /usr/local/lib/node_modules/bower/lib/commands/index.js:48:20
    at _rejected (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:844:24)
    at /usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:870:30
    at Promise.when (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:1122:31)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/bower/lib/node_modules/q/q.js:788:41)
System info:
Bower version: 1.8.0
Node version: 6.5.0
OS: Darwin 16.1.0 x64

It's all terribly confusing for a flask newb... If you could explain it I could write some docs.

**Comment by wroberts** [comment]
In #183 I've pushed some changes that I used in [a recent project of mine](https://github.com/wroberts/annotator), and I thought I would write here in case anyone wants to take a look.

This patch includes replacing Flask-Assets with webpack; the setup produces css and js bundles, and quickstart is documented in the README.  For development, there is a webpack-dev-server that can run beside Flask and does hot updates of js and css.  Also, webpack runs babel so that javascript can be written in ES6.
